stacker/blueprints/rds/base.py

```diff
@@ -0,0 +1,336 @@
+            "MultiAZ": {
+                "type": "String",
+                "description": "Set to 'false' to disable MultiAZ support.",
+                "default": "true"},
+        }
+        engine_versions = self.get_engine_versions()
+        if engine_versions:
+            master_parameters['EngineVersion']['allowed_values'] = \
+                engine_versions
+
+        if not self.ENGINE:
+            master_parameters['Engine'] = {
+                "type": "String",
+                "description": "Database engine for the RDS Instance.",
+                "allowed_values": RDS_ENGINES
+            }
+        else:
+            if self.ENGINE not in RDS_ENGINES:
+                raise ValueError("ENGINE must be one of: %s" %
+                                 ", ".join(RDS_ENGINES))
+
+        # Merge common parameters w/ master only parameters
+        parameters = common_parameters().update(master_parameters)
+
+        return parameters
+
+    def create_conditions(self):
+        self.template.add_condition(
+            "HasInternalZone",
+            Not(Equals(Ref("InternalZoneId"), "")))
+        self.template.add_condition(
+            "HasInternalZoneName",
+            Not(Equals(Ref("InternalZoneName"), "")))
+        self.template.add_condition(
+            "HasInternalHostname",
+            Not(Equals(Ref("InternalHostname"), "")))
+        self.template.add_condition(
+            "CreateInternalHostname",
+            And(Condition("HasInternalZone"),
+                Condition("HasInternalZoneName"),
+                Condition("HasInternalHostname")))
+        self.template.add_condition(
+            "ProvisionedIOPS",
+            Not(Equals(Ref("IOPS"), "0")))
+        self.template.add_condition(
+            "NoProvisionedIOPS",
+            Equals(Ref("IOPS"), "0"))
+
+    def create_subnet_group(self):
+        t = self.template
+        t.add_resource(
+            DBSubnetGroup(
+                SUBNET_GROUP,
+                DBSubnetGroupDescription="%s VPC subnet group." % self.name,
+                SubnetIds=Ref("PrivateSubnets")))
+
+    def create_security_group(self):
+        t = self.template
+        sg = t.add_resource(
+            ec2.SecurityGroup(
+                SECURITY_GROUP,
+                GroupDescription="%s RDS security group" % self.name,
+                VpcId=Ref("VpcId")))
+        t.add_output(Output("SecurityGroup", Value=Ref(sg)))
+
+    def get_db_instance(self):
+        return If("ProvisionedIOPS", DBINSTANCE_WITH_IOPS,
+                  DBINSTANCE_NO_IOPS)
+
+    def get_db_endpoint(self):
+        endpoint = GetAtt(self.get_db_instance(), "Endpoint.Address")
+        return endpoint
+
+    def create_rds(self):
+        t = self.template
+        # Non-provisioned iops database
+        t.add_resource(
+            DBInstance(
+                DBINSTANCE_NO_IOPS,
+                Condition="NoProvisionedIOPS",
```

Comment:
i would pull out all of these common parameters so i don't have to duplicate specifying them in each